src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -1058,31 +1321,124 @@ def _set_up_defaults(
+                if custom_location_from_env is None or custom_location.connected_cluster_id.lower() != custom_location_from_env.host_resource_id.lower():
+                    continue
+            env_list.append(e)
+
+        if len(env_list) == 1:
+            resource_group.name = parse_resource_id(env_list[0]["id"])["resource_group"]
+            env.set_name(env_list[0]["id"])
+            env.custom_location_id = env_list[0]["extendedLocation"]["name"]
+        if len(env_list) > 1:
+            if env.name:
+                raise ValidationError(
+                    f"There are multiple environments with name {env.name} on the subscription. "
+                    "Please specify which resource group your Connected environment is in."
+                )  # get ACR details from --image, if possible
+            else:
+                resource_group.name = parse_resource_id(env_list[0]["id"])["resource_group"]
+                env.set_name(env_list[0]["id"])
+                env.custom_location_id = env_list[0]["extendedLocation"]["name"]
+
+
+def _infer_existing_custom_location_or_extension(
+        cmd,
+        name,
+        location,
+        resource_group: "ResourceGroup",
+        env: "ContainerAppEnvironment",
+        custom_location: "CustomLocation",
+        extension: "Extension"
+):
+    # Try to get existed custom location and extension
+    # Set up default values for not existed resources(env, custom location, extension)
+    if env.is_connected_environment():
+        if not env.check_exists():
+            _prepare_custom_location_args(cmd, custom_location=custom_location, extension=extension)
+            if location is None:
+                env.location = custom_location.location
+                resource_group.location = custom_location.location
+            if custom_location.exists:
+                env.custom_location_id = custom_location.get_rid()
+            else:
+                random_int = randint(0, 9999)
+                resource_group.name = get_randomized_name(get_profile_username(), name=resource_group.name, random_int=random_int)
+                env.name = env.name if env.name else "{}-env".format(name).replace("_", "-")
+                custom_location.name = get_randomized_name_with_dash(prefix=get_profile_username(), name=custom_location.name, initial="env-location", random_int=random_int)
+                custom_location.resource_group_name = resource_group.name
+                env.custom_location_id = custom_location.get_rid()
+                # If not existed extension, set up values for creating
+                if not extension.exists:
+                    extension.name = 'containerapp-ext'
+                    extension.namespace = 'containerapp-ns'
+                    extension.logs_rg = resource_group.name
+                    extension.logs_location = resource_group.location
```

Comment:
is log required for extension?

Reply:
A Log Analytics workspace is optional, but recommended.
 It is also consistent with managed environment.